pkgs/by-name/sc/scipopt-papilo/package.nix

```
@@ -0,0 +1,49 @@
+{
+  lib,
+  stdenv,
+  cmake,
+  fetchFromGitHub,
+  boost,
+  blas,
+  gmp,
+  tbb_2021_11,
+  gfortran,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "scipopt-papilo";
+  version = "2.4.1";
+
+  src = fetchFromGitHub {
+    owner = "scipopt";
+    repo = "papilo";
+    tag = "v${version}";
+    hash = "sha256-oQ9iq5UkFK0ghUx6uxdJIOo5niQjniHegSZptqi2fgE=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [
+    blas
+    gmp
+    gfortran
+    boost
+    tbb_2021_11
+  ];
+
+  cmakeFlags = [
+    (lib.cmakeBool "GMP" true)
+    (lib.cmakeBool "QUADMATH" true)
+    (lib.cmakeBool "TBB" true)
+    (lib.cmakeBool "TBB_DOWNLOAD" false)
+    (lib.cmakeBool "SOPLEX" false)
+  ];
+
```

```suggestion
  doCheck = true;
```
This one also has a test suite we should run as part of the package.